tests/test_build.py

```diff
@@ -787,3 +787,62 @@ def test_empty_templates(sphinx_build_factory):
     sphinx_build = sphinx_build_factory("base", confoverrides=confoverrides).build()
     toc_items = sphinx_build.html_tree("page1.html").select(".toc-item")
     assert not any(ii.select(".tocsection.sourcelink") for ii in toc_items)
+
+
+def test_translations(sphinx_build_factory):
+    """Test that basic translation functionality works.
+
+    This will build our test site with the French language, and test
+    that a few phrases are in French.
+
+    TODO: At first, we expect some of these phrases to be *incorrectly* in
+    English. This is because we haven't added translation files for them.
+    We should change this test to the appropriate French versions once we add
+    support for other languages.
+
+    Then, we use this test to catch regressions if we change wording without
+    changing the translation files."""
+
+    confoverrides = {
+        "language": "fr",
+        "html_context": {
+            "github_user": "pydata",
+            "github_repo": "pydata-sphinx-theme",
+            "github_version": "main",
+        },
+        "html_theme_options": {
+            "use_edit_page_button": True,
+        },
+    }
+    sphinx_build = sphinx_build_factory("base", confoverrides=confoverrides).build()
+
+    # Use a section page so that we have section navigation in the sidebar
+    index = sphinx_build.html_tree("section1/index.html")
+
+    # TODO: Add translations where there are english phrases below
+    sidebar_primary = index.select(".bd-sidebar-primary")[0]
+    assert "Site Navigation" in str(sidebar_primary)
+    assert "Section Navigation" in str(sidebar_primary)
+
+    # TODO: Add translations where there are english phrases below
+    sidebar_secondary = index.select(".bd-sidebar-secondary")[0]
+    assert "Montrer le code source" in str(sidebar_secondary)
```

Comment:
Is this a translation that you came up with or is it from sphinx?
I'm a native speaker and that sounds very weird compared to "Voir code source"

Reply:
I believe this is just what sphinx spat out - was not my translation! Another good reason to add out own translations? 🙂